src/lib/dataviews/formula/FormulaImpl.ts

```diff
@@ -0,0 +1,22 @@
+import { AggregationType } from '../../data/operations/aggregation/aggregation';
+import { DataViewData, DataViewModeBase } from '../mode/DataViewModeBase';
+import { DataViewImplBase } from '../DataViewImplBase';
+
+export class FormulaImpl<T extends DataViewModeBase> extends DataViewImplBase<T> {
+  public async getData(): Promise<Partial<DataViewData>> {
+    let formulaResponse;
+
+    try {
+      formulaResponse = await this.dataView.formula(this.operation);
+    } catch (error) {
+      this.emit('error', [error]);
```

Comment:
Should we emit errors or just throw....?

Reply:
I copied the [carto.js behaviour](https://carto.com/developers/carto-js/examples/#example-category-widget) but not sure about emitting or throwing